src/lender/tranche.sol

```diff
@@ -200,18 +200,23 @@ contract Tranche is Math, Auth, FixedPoint {
     // the disburse function can be used after an epoch is over to receive currency and tokens
     function disburse(address usr,  uint endEpoch) public auth returns (uint payoutCurrencyAmount, uint payoutTokenAmount, uint remainingSupplyCurrency, uint remainingRedeemToken) {
         require(users[usr].orderedInEpoch <= epochTicker.lastEpochExecuted());
+
+        uint lastEpochExecuted = epochTicker.lastEpochExecuted();
+
+        if (endEpoch > lastEpochExecuted) {
+            // it is only possible to disburse epochs which are already over
+            endEpoch = lastEpochExecuted;
+        }
```

Comment:
This was another problem because this check previously only happened in the calcDisburse method.

It would allow an investor to use a endEpoch higher than current epoch the disburse calculation would fine but we would overwrite this variable in `line 213`